src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -120,62 +148,30 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
                         _documentToLastReportedInformation.TryRemove(docId, out _);
                 }
 
-                // Handle the priority doc first.
-                var priorityDocument = solution.GetDocument(priorityDocumentId);
-                if (priorityDocument != null)
-                    await ProcessProjectAsync(priorityDocument.Project, priorityDocument, useFrozenSnapshots, callback, cancellationToken).ConfigureAwait(false);
-
-                // Wait a little after the priority document and process the rest at a lower priority.
-                await _listener.Delay(DelayTimeSpan.NonFocus, cancellationToken).ConfigureAwait(false);
-
                 // Process the rest of the projects in dependency order so that their data is ready when we hit the 
                 // projects that depend on them.
-                var dependencyGraph = solution.GetProjectDependencyGraph();
+                var dependencyGraph = frozenSolution.GetProjectDependencyGraph();
                 foreach (var projectId in dependencyGraph.GetTopologicallySortedProjects(cancellationToken))
-                {
-                    if (projectId != priorityDocumentId?.ProjectId)
-                        await ProcessProjectAsync(solution.GetRequiredProject(projectId), specificDocument: null, useFrozenSnapshots, callback, cancellationToken).ConfigureAwait(false);
-                }
+                    await ProcessProjectAsync(frozenSolution.GetRequiredProject(projectId), callback, cancellationToken).ConfigureAwait(false);
             }
```

Comment:
the full solution walk is now simpler as it just processes projects one at a time without having to care about a priority doc.